distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -273,6 +273,9 @@ get_constructor_name atom_constructor = @Builtin_Method "Meta.get_constructor_na
 new_atom : Any -> Array -> Atom
 new_atom constructor fields = @Builtin_Method "Meta.new_atom"
 
```

Comment:
I am not really happy to be introducing new `lazy_atom` method, neither I am satisfied with its name. An alternative approach would be to modify `new_atom` method to take `lazy_index=-1` argument. Prototype is available in [wip/jtulach/ReuseMetaNewAtom_183578531](https://github.com/enso-org/enso/compare/wip/jtulach/NoSetAtomField_183578531...JaroslavTulach:enso:wip/jtulach/ReuseMetaNewAtom_183578531) branch. This approach places the _initialization function_ directly into the `Atom` - that may not be desirable. As such I am just offering the link as an alternative, not real proposal.